client/shared/src/main/scala/org/http4s/client/middleware/Retry.scala

```diff
@@ -38,6 +39,8 @@ import scala.math.random
 object Retry {
   private[this] val logger = getLogger
 
+  val RetryCountKey: Key[Int] = Key.newKey[cats.effect.SyncIO, Int].unsafeRunSync()
```

Comment:
Maybe we need a brief doc here. Is this the attempt we are on (e.g. 1 for first, 2 for second) or the number of previous attempts (0 for first, 1 for second)?

Reply:
So, the initial request should set this key to `0`? Because I'm not sure if that's what it currently does, I think it starts from 1.
https://github.com/http4s/http4s/blob/331a588be0c4bbdf5672420e06329ce6e23757db/client/shared/src/main/scala/org/http4s/client/middleware/Retry.scala#L128

Reply:
In my case, which is the open telemetry standard. I think this should just be set to the number http4s uses and other should adjust based on that.

Reply:
Sounds good to me. Sorry, one last bikeshed: would `AttemptCount` make more sense?

Reply:
I like it, more clear. Will switch.